src/core/scene/a-scene.js

```diff
@@ -492,6 +492,11 @@ module.exports.AScene = registerElement('a-scene', {
         });
         renderer.setPixelRatio(window.devicePixelRatio);
         renderer.sortObjects = false;
+        // We expect camera-set-active to be triggered at least once in the life of an a-frame app. Usually soon after
+        // it is initialized.
+        this.addEventListener('camera-set-active', function () {
+          renderer.vr.setPoseTarget(self.camera.el.object3D);
```

Comment:
This should only be done if in VR mode? Need to account for inspector and spectator camera and such

Reply:
I could put a check in but I don't think it's necessary. `setPoseTarget` has no adverse effects in non-vr mode and it makes sense to keep the active camera up to date for when the user chooses to enter VR mode.

Reply:
For spectator camera, the active camera is not the VR camera/pose target

Reply:
Hmm. Is there an existing example of the spectator camera in use that I could test with?

Reply:
As far as I can tell, the spectator camera is never assigned to `sceneEl.camera`. "spectator camera" and "active camera" are different concepts, aren't they?

Reply:
@brianpeiris Yes they are different. Look at this example:

https://glitch.com/edit/#!/probable-sausage?path=index.html:14:16

The camera system has a tock method to render the scene using the spectatorCamera:

https://github.com/aframevr/aframe/blob/master/src/systems/camera.js#L148

When entering VR the camera on `sceneEl.camera` is the one used in the headset. We should probably only use the spectator camera in VR mode and honor the normal one in 2D mode.

Reply:
Right, so I think we're okay here. Thanks for the merge.

Reply:
Thanks!